pkg/omni/fetcher/fetcher_test.go

```diff
@@ -0,0 +1,145 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fetcher_test
+
+import (
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/vulcanize/vulcanizedb/examples/constants"
+	"github.com/vulcanize/vulcanizedb/pkg/geth"
+	"github.com/vulcanize/vulcanizedb/pkg/geth/client"
+	rpc2 "github.com/vulcanize/vulcanizedb/pkg/geth/converters/rpc"
+	"github.com/vulcanize/vulcanizedb/pkg/geth/node"
+	"github.com/vulcanize/vulcanizedb/pkg/omni/fetcher"
+)
+
+var _ = Describe("Fetcher Test", func() {
+	blockNumber := int64(6194634)
+	var realFetcher fetcher.Fetcher
+
+	BeforeEach(func() {
+		infuraIPC := "https://mainnet.infura.io/v3/b09888c1113640cc9ab42750ce750c05"
+		rawRpcClient, err := rpc.Dial(infuraIPC)
+		Expect(err).NotTo(HaveOccurred())
+		rpcClient := client.NewRpcClient(rawRpcClient, infuraIPC)
+		ethClient := ethclient.NewClient(rawRpcClient)
+		blockChainClient := client.NewEthClient(ethClient)
+		node := node.MakeNode(rpcClient)
+		transactionConverter := rpc2.NewRpcTransactionConverter(ethClient)
+		blockChain := geth.NewBlockChain(blockChainClient, node, transactionConverter)
+		realFetcher = fetcher.NewFetcher(blockChain)
+	})
+
+	Describe("Fetch big.Int test", func() {
+
+		It("fetch totalSupply big.Int", func() {
+			bigInt, err := realFetcher.FetchBigInt("totalSupply", constants.DaiAbiString, constants.DaiContractAddress, blockNumber, nil)
+			Expect(err).NotTo(HaveOccurred())
+			expectedBigInt := big.Int{}
+			expectedBigInt.SetString("47327413946297204537985606", 10)
+			Expect(bigInt.String()).To(Equal(expectedBigInt.String()))
+		})
+
+		It("returns an error if the call to the blockchain fails", func() {
+			result, err := realFetcher.FetchBigInt("totalSupply", "", "", 0, nil)
+
+			Expect(result).To(Equal(big.Int{}))
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("totalSupply"))
+		})
+	})
+
+	Describe("Fetch bool test", func() {
+
+		It("fetch stopped boolean", func() {
+			boo, err := realFetcher.FetchBool("stopped", constants.DaiAbiString, constants.DaiContractAddress, blockNumber, nil)
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(boo).To(Equal(false))
+		})
+
+		It("returns an error if the call to the blockchain fails", func() {
+			boo, err := realFetcher.FetchBool("stopped", "", "", 0, nil)
+
+			Expect(boo).To(Equal(false))
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(ContainSubstring("stopped"))
+		})
+	})
+
+	Describe("Fetch address test", func() {
+
+		It("fetch owner address", func() {
+			expectedAddr := common.HexToAddress("0x0000000000000000000000000000000000000000")
```

Comment:
wondering whether we might want to test against another address to be sure we aren't getting zero values on the fetch

Reply:
Good call, TrueUSD has a non-zero owner- I will add that.